optuna/study.py

```diff
@@ -598,35 +652,45 @@ def _ask(self) -> trial_module.Trial:
             trial_id = self._storage.create_new_trial(self._study_id)
         return trial_module.Trial(self, trial_id)
 
-    def _tell(self, trial: trial_module.Trial, state: TrialState, value: Optional[float]) -> None:
+    def _tell(
+        self, trial: trial_module.Trial, state: TrialState, values: Optional[List[float]]
+    ) -> None:
         if state == TrialState.COMPLETE:
-            assert value is not None
-        if value is not None:
-            self._storage.set_trial_value(trial._trial_id, value)
+            assert values is not None
+        if values is not None:
+            self._storage.set_trial_values(trial._trial_id, values)
         self._storage.set_trial_state(trial._trial_id, state)
 
-    def _log_completed_trial(self, trial: trial_module.Trial, value: float) -> None:
-        # This method is overwritten by `MultiObjectiveStudy` using `types.MethodType` so one must
-        # be careful modifying this method, e.g. making this a free function.
+    def _log_completed_trial(
+        self, trial: trial_module.Trial, value: Union[float, Sequence[float]]
+    ) -> None:
 
         if not _logger.isEnabledFor(logging.INFO):
             return
 
-        _logger.info(
-            "Trial {} finished with value: {} and parameters: {}. "
-            "Best is trial {} with value: {}.".format(
-                trial.number, value, trial.params, self.best_trial.number, self.best_value
+        if isinstance(value, Sequence):
+            _logger.info(
+                "Trial {} finished with value: {} and parameters: {}. ".format(
+                    trial.number, value, trial.params
+                )
+            )
+        else:
+            _logger.info(
+                "Trial {} finished with value: {} and parameters: {}. "
+                "Best is trial {} with value: {}.".format(
+                    trial.number, value, trial.params, self.best_trial.number, self.best_value
+                )
             )
-        )
 
 
 def create_study(
     storage: Optional[Union[str, storages.BaseStorage]] = None,
     sampler: Optional["samplers.BaseSampler"] = None,
     pruner: Optional[pruners.BasePruner] = None,
     study_name: Optional[str] = None,
-    direction: str = "minimize",
+    direction: Optional[str] = None,
     load_if_exists: bool = False,
+    directions: Optional[Sequence[str]] = None,
```

Comment:
I'm not sure about this API. Although it can be explained through the docs, it's not too pretty 🤔  Do you think it's be strange if we make `directions` a union and deprecate the old one?

Reply:
We had a private discussion about this comment in our team. As a result, we have decided to proceed with the current PR. Please let me resolve this conversation.